app/src/main/java/net/bible/service/db/DatabaseContainer.kt

```diff
@@ -551,6 +551,30 @@ private val SQUASH_30_33 = object : Migration(30, 33) {
     }
 }
 
+private val MIGRATION_33_34_Bookmarks = object : Migration(33, 34) {
+    override fun doMigrate(db: SupportSQLiteDatabase) {
+        db.apply {
+            execSQL("PRAGMA foreign_keys=OFF;")
+            execSQL("ALTER TABLE bookmark RENAME TO bookmark_old;")
+            execSQL("ALTER TABLE label RENAME TO label_old;")
+
+            execSQL("CREATE TABLE IF NOT EXISTS `Bookmark` (`id` INTEGER NOT NULL, `createdOn` INTEGER, `key` TEXT NOT NULL, `versification` TEXT, `playbackSettings` TEXT, PRIMARY KEY(`id`))")
+            execSQL("CREATE TABLE IF NOT EXISTS `Label` (`id` INTEGER NOT NULL, `name` TEXT NOT NULL, `bookmarkStyle` TEXT, PRIMARY KEY(`id`))")
+            execSQL("CREATE TABLE IF NOT EXISTS `BookmarkToLabel` (`bookmarkId` INTEGER NOT NULL, `labelId` INTEGER NOT NULL, PRIMARY KEY(`bookmarkId`, `labelId`), FOREIGN KEY(`bookmarkId`) REFERENCES `Bookmark`(`id`) ON UPDATE NO ACTION ON DELETE CASCADE , FOREIGN KEY(`labelId`) REFERENCES `Label`(`id`) ON UPDATE NO ACTION ON DELETE CASCADE )")
+            execSQL("CREATE INDEX IF NOT EXISTS `index_BookmarkToLabel_labelId` ON `BookmarkToLabel` (`labelId`)")
+
+            execSQL("INSERT INTO Bookmark SELECT * from bookmark_old;")
+            execSQL("INSERT INTO Label SELECT * from label_old;")
+            execSQL("INSERT INTO BookmarkToLabel SELECT * from bookmark_label;")
+
+            execSQL("DROP TABLE bookmark_old;")
+            execSQL("DROP TABLE label_old;")
+            execSQL("DROP TABLE bookmark_label;")
+            execSQL("PRAGMA foreign_keys=ON;")
```

Comment:
Possibly we should stop using the `PRAGMA foreign_keys=ON or OFF` to prevent further issues of https://github.com/AndBible/and-bible/issues/770#issuecomment-658889797

I don't think it will be an issue here, but since Sqlite 3.26+ operates different on foreign keys with table rename, if we would stop using `PRAGMA foreign_keys` then we would find the issue with earlier Sqlite versions too (i.e. then earlier versions of Sqlite would operate the same way as 3.26+).

Reply:
Okay here it indeed works without disabling foreign keys. So removed it.

Reply:
My point is that after sqlite 3.26+ it is useless to disable foreign keys and all it does is possibly cause it to work for < 3.26 but give issues for 3.26+. So in my opinion we should make it work with disabling foreign keys from here on, so that there will not be the issue at 3.26+

Reply:
So we should actually remove use of it from all migrations, rather than just this one?
But sometimes it is needed to remove foreign keys handling when migrating data, isn't it? 
Especially if the data has links to the same table or if two tables have both links to each other.

Reply:
I would not remove from previous migrations.

You're right, I did not think about all circumstances. For example, when changing data on only 1 table that has links from other tables it could be needed.

It's just the way foreign keys are handled differently after 3.26 that we need to keep in mind for migrations, but probably won't usually be an issue.